aspnet/mobile/native-mobile-backend.rst

```diff
@@ -1,11 +1,177 @@
+
+.. image:: native-mobile-backend/_static/todo-android.png
+
+Features
+^^^^^^^^
+
+The TodoRest app supports listing, adding, deleting, and updating To-Do items. Each item has an ID, a Name, Notes, and a property indicating whether it's been Done yet.
+
+The main view of the items, as shown above, lists each item's name and indicates if it is done with a checkmark.
+
+Tapping the ``+`` icon opens an add item dialog:
+
+.. image:: native-mobile-backend/_static/todo-android-new-item.png
+
+Tapping an item on the main list screen opens up an edit dialog where the item's Name, Notes, and Done settings can be modified, or the item can be deleted:
+
+.. image:: native-mobile-backend/_static/todo-android-edit-item.png
+
+This sample is configured by default to use backend services hosted at developer.xamarin.com, which allow read-only operations. To test it out yourself against the ASP.NET Core app created in the next section running on your computer, you'll need to update the app's ``RestUrl`` constant. Navigate to the ``TodoREST`` project and open the `Constants.cs` file. Replace the ``RestUrl`` with a URL that includes your machine's IP address (not localhost or 127.0.0.1, since this address is used from the device emulator, not from your machine). Include the port number as well (5000). In order to test that your services work with a device, ensure you don't have an active firewall blocking access to this port.
+
+.. code-block:: csharp
+
+    // URL of REST service (Xamarin ReadOnly Service)
+    //public static string RestUrl = "http://developer.xamarin.com:8081/api/todoitems{0}";
+    
+    // use your machine's IP address
+    public static string RestUrl = "http://192.168.1.207:5000/api/todoitems/{0}";
+
+
+Creating the ASP.NET Core Project
+---------------------------------
+
+Create a new ASP.NET Core MVC project in Visual Studio. Choose the Web API template and No Authentication.
+
+.. image:: native-mobile-backend/_static/web-api-template.png
+
+The application should respond to all requests made to port 5000. Update `Program.cs` to include ``.UseUrls("http://*:5000")`` to achieve this:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Program.cs
+  :language: c#
+  :lines: 10-16
+  :dedent: 12
+  :emphasize-lines: 3
+
+.. note:: Make sure you run the application directly, rather than behind IIS Express, which ignores non-local requests by default. Run ``dotnet run`` from the command line, or choose the application name profile from the debug menu in Visual Studio.
+
+Add a model class to represent To-Do items. Mark required fields using the ``[Required]`` attribute:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Models/TodoItem.cs
```

Comment:
Agreed. Not sure if it's worth fixing but I'll see if I have time to take care of it with this next commit.